source/common/common/assert.h

```diff
@@ -3,21 +3,34 @@
 #include "common/common/logger.h"
 
 namespace Envoy {
+
 /**
  * assert macro that uses our builtin logging which gives us thread ID and can log to various
  * sinks.
+ *
+ * The old style release assert was of the form RELEASE_ASSERT(foo == bar);
+ * where it would log stack traces and the failed conditional and crash if the
+ * condition is not met.  The are many legacy RELEASE_ASSERTS in Envoy which
+ * were converted to RELEASE_ASSERT(foo == bar, "");
+ *
+ * The new style of release assert is of the form
+ * RELEASE_ASSERT(foo == bar, "reason foo should actually be bar");
+ * new uses of RELEASE_ASSERT are strongly encouraged to supply a verbose
+ * explanation of what went wrong.
  */
-#define RELEASE_ASSERT(X)                                                                          \
+#define RELEASE_ASSERT(X, DETAILS)                                                                 \
   do {                                                                                             \
     if (!(X)) {                                                                                    \
+      std::string details = DETAILS;                                                               \
```

Comment:
Could this be an absl::string_view to avoid copying? Or does that break things?

Reply:
Why do you want to optimize copy on a termination path?!

Reply:
I would recommend writing this as `const std::string& details = (DETAILS);`

Reply:
Ha, good point; I think that comment just came out of habit.